src/components/beat.js

```diff
@@ -421,13 +434,25 @@ AFRAME.registerComponent('beat', {
       if (this.beatSystem.data.gameMode === 'punch') {
         bbox.setFromObject(weaponEl.getObject3D('mesh')).expandByScalar(0.1);
         weaponBbox = bbox;
+
+        if (!weaponBbox) { continue; }
+        if (!weaponBbox.intersectsBox(beatBbox)) { continue; }
       } else {
         const blade = weaponEl.components.blade;
-        weaponBbox = blade.boundingBox;
-      }
 
-      if (!weaponBbox) { continue; }
-      if (!weaponBbox.intersectsBox(beatBbox)) { continue; }
+        for (let i = 0; i < 4; i++) {
+          this.bladeLocalPositions [i].copy(blade.bladeWorldPositions [i]);
+          this.blockEl.object3D.worldToLocal(this.bladeLocalPositions [i]);
+        }
+
+        this.bladeLocalTriangles [0].set(this.bladeLocalPositions [0], this.bladeLocalPositions [1], this.bladeLocalPositions [2]);
+        this.bladeLocalTriangles [1].set(this.bladeLocalPositions [2], this.bladeLocalPositions [3], this.bladeLocalPositions [0]);
+
+        const beatBbox = this.blockEl.getObject3D('mesh').geometry.boundingBox;
+        if (!beatBbox.intersectsTriangle(this.bladeLocalTriangles [0]) && !beatBbox.intersectsTriangle(this.bladeLocalTriangles [0])) {
```

Comment:
i think the second one should be `[1]`?

Reply:
You are absolutely right, sorry, fixing that...